### PR TITLE
pkcs11: test mechanism info

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: host/xtest/pkcs11_1000.c [hunk 1]
```diff
@@ -146,6 +151,92 @@ static void xtest_tee_test_1001(ADBG_Case_t *c)
...
+				continue;
+
+			pos = snprintf(&log[0], sizeof(log),
+				       "%-30s Key size [%03lu %03lu]",
+				       ckm2str(mecha_types[j]),
+				       mecha_info.ulMinKeySize,
+				       mecha_info.ulMaxKeySize);
+
+			if (pos > sizeof(log)) {
+				Do_ADBG_Log("| Error: internal short buffer");
```
Comment (jenswi-linaro) [inline host/xtest/pkcs11_1000.c]: Did the test pass?

File: host/xtest/pkcs11_1000.c [hunk 2]
```diff
@@ -146,6 +151,92 @@ static void xtest_tee_test_1001(ADBG_Case_t *c)
...
+			goto out;
+
+		if (level)
+			Do_ADBG_Log("Token #%zu mechanism capabilities:", i);
+
+		for (j = 0; j < mecha_count; j++) {
+			size_t pos = 0;
+			size_t k = 0;
+			/* 1024byte should be enough, if not truncates  */
+			char log[1024] = { 0 };
```
Comment (jenswi-linaro) [inline host/xtest/pkcs11_1000.c]: The name `log` may conflict with the standard function `log()`.

File: host/xtest/pkcs11_1000.c [hunk 3]
```diff
@@ -146,6 +151,92 @@ static void xtest_tee_test_1001(ADBG_Case_t *c)
...
+
+			rv = C_GetMechanismInfo(slot, mecha_types[j], &mecha_info);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto out;
+
+			/*  Verbose output on high levels */
+			if (level == 0)
+				continue;
+
+			pos = snprintf(&log[0], sizeof(log),
```
Comment (jenswi-linaro) [inline host/xtest/pkcs11_1000.c]: `&log[0]` looks a bit funny, how about just `log`?

File: host/xtest/pkcs11_1000.c [hunk 4]
```diff
@@ -146,6 +151,92 @@ static void xtest_tee_test_1001(ADBG_Case_t *c)
 	}
 
 	Do_ADBG_EndSubCase(c, "Test C_Get{Slot|Token}Info()");
+	Do_ADBG_BeginSubCase(c, "Test C_GetMechanism{List|Info}()");
+
+	for (i = 0; i < slot_count; i++) {
+		CK_SLOT_ID slot = slot_ids[i];
```
Comment (jenswi-linaro) [inline host/xtest/pkcs11_1000.c]: What's in this loop should probably go into a helper function instead.

File: host/xtest/pkcs11_1000.c (hunk 2 above)
Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: ok, will change. thanks.

File: host/xtest/pkcs11_1000.c (hunk 3 above)
Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: sorry, will fix, here and there.

File: host/xtest/pkcs11_1000.c (hunk 1 above)
Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: :)  I will remove the test.

File: host/xtest/pkcs11_1000.c (hunk 4 above)
Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: Ok I you think it helps.

File: host/xtest/pkcs11_1000.c [hunk 5]
```diff
@@ -50,6 +50,75 @@ static void xtest_tee_test_1000(ADBG_Case_t *c)
...
+}
+
+static int check_mechanism_list(ADBG_Case_t *c, CK_SLOT_ID slot,
+				CK_MECHANISM_TYPE_PTR mecha_types,
+				CK_ULONG mecha_count)
+{
+	size_t n = 0;
+
+	/* Verbose output on high levels */
+	if (level)
```
Comment (jenswi-linaro) [inline host/xtest/pkcs11_1000.c]: Level is not log level, it's how much testing is done. See xtest_tee_test_4006() for instance.

Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: I agree. But there is no verbosity level defined. I assume test level > 0 means a bit more than the min testing hence adding a trace is not that noisy.

Comment (etienne-lms) [reply host/xtest/pkcs11_1000.c]: I'll discard this log stuff.

**Comment by etienne-lms** [comment]
I removed the commit "pkcs11 1001: print mechanism info upon test level" as per https://github.com/OP-TEE/optee_test/pull/403#discussion_r391842907.

**Comment by etienne-lms** [comment]
fixup commits squashed.
tag applied.
thanks.

**Pull request merged**